src/shared/services/activity/activity.service.ts

```diff
@@ -47,23 +47,34 @@ export class ActivityService {
   }
 
   public getActivities(...caseId: string[]): Observable<Activity[]> {
-    const options = this.getOptions();
-    const url = this.activityUrl() + `/cases/${caseId.join(',')}/activity`;
-    return this.http
-      .get(url, options, false, ActivityService.handleHttpError)
-      .map(response => response);
+    try {
+      const options = this.getOptions();
+      const url = this.activityUrl() + `/cases/${caseId.join(',')}/activity`;
+      return this.http
+        .get(url, options, false, ActivityService.handleHttpError)
+        .map(response => response);
+    } catch (error) {
+      console.log('user may not be authenticated.' + error);
+    }
   }
 
   public postActivity(caseId: string, activity: string): Observable<Activity[]> {
-    const options = this.getOptions();
-    const url = this.activityUrl() + `/cases/${caseId}/activity`;
-    let body = { activity };
-    return this.http
-      .post(url, body, options, false)
-      .map(response => response);
+    try {
+      const options = this.getOptions();
+      const url = this.activityUrl() + `/cases/${caseId}/activity`;
+      let body = { activity };
+      return this.http
+        .post(url, body, options, false)
+        .map(response => response);
+    } catch (error) {
+      console.log('user may not be authenticated.' + error);
+    }
   }
 
   public verifyUserIsAuthorized(): void {
+    if (this.sessionStorageService.getItem('userDetails') === undefined) {
```

Comment:
Why are  we checking explicitly for undefined. Sessionstorage return null if userDetails is not in local storage.  
```suggestion
    if (!this.sessionStorageService.getItem('userDetails')) {
```

Reply:
@kuda1992 I tried using that before I changes it to this. I also not sure why its behaving like this

Reply:
@RiteshHMCTS I don't get what you're trying to say. What's behaving like this?

Reply:
Thanks for getting back @RiteshHMCTS I will approve